### PR TITLE
use ARG to allow for a 3.6 and 3.7 docker image from the same Dockerfile

### PR DESCRIPTION
**Pull request by escapewindow**

### CODE REVIEW

**Comment by escapewindow** [comment]
Ok, I think I finally got the README formatting right.

File: README.rst [hunk 1]
```diff
@@ -67,19 +79,20 @@ then
...
 GPG Homedir testing
 ^^^^^^^^^^^^^^^^^^^
 
-Sometimes it's nice to be able to test things like ``rebuild_gpg_homedirs``.  To do so::
+Sometimes it's nice to be able to test things like ``rebuild_gpg_homedirs``.  To do so, `build the docker image`_, then::
 
-    PY_VERSION=37  # or 36
-    cp docker/Dockerfile.test-py$PY_VERSION Dockerfile
-    docker build -t scriptworker-test . && docker run -i scriptworker-test bash -il
+    docker run -i scriptworker-test bash -il
```
Comment (tomprince) [inline README.rst]:
```suggestion
    docker run -i scriptworker-test-$PY_DOT_VERSION bash -il
```

File: README.rst [hunk 2]
```diff
@@ -67,19 +79,20 @@ then
 
 ``tox``
 
-It's also possible to create a ``./secrets.json`` as above, then::
+It's also possible to create a ``./secrets.json`` as above, and `build the docker image`_. Then::
```
Comment (tomprince) [inline README.rst]: It appears that creating a secrets.json is *required* to build the docker image.

File: README.rst [hunk 3]
```diff
@@ -34,6 +34,18 @@ Credentials can live in ``./scriptworker.yaml``, ``./secrets.json``, ``~/.script
...
+-------
+Building a docker image
+-------
+
+First, create a `secrets.json`. For integration testing, you'll need to define the `integration_credentials`; to do any other authenticated work, you'll need to define `credentials`.
+
+Then::
+
+    PY_DOT_VERSION=3.7  # or 3.6
+    cp docker/Dockerfile.test Dockerfile
```
Comment (tomprince) [inline README.rst]:
```suggestion
```

File: README.rst [hunk 4]
```diff
@@ -34,6 +34,18 @@ Credentials can live in ``./scriptworker.yaml``, ``./secrets.json``, ``~/.script
...
+    docker build -t scriptworker-test-$PY_DOT_VERSION --build-arg PY_DOT_VERSION=$PY_DOT_VERSION .
```
Comment (tomprince) [inline README.rst]: You can specify the docker file to use, without having to copy it.
```suggestion
    docker build -t scriptworker-test-$PY_DOT_VERSION --build-arg PY_DOT_VERSION=$PY_DOT_VERSION --file docker/Dockerfile.test .
```

**Comment by escapewindow** [comment]
Thanks! I imagine this will continue to evolve.

**Pull request merged**